testutil/daemon/daemon.go

```diff
@@ -91,6 +97,63 @@ type Daemon struct {
 	CachedInfo types.Info
 }
 
+// NewDaemon returns a Daemon instance to be used for testing.
+// This will create a directory such as d123456789 in the folder specified by
+// $DOCKER_INTEGRATION_DAEMON_DEST or $DEST.  The daemon will not automatically
+// start.
+func NewDaemon(ops ...func(*Daemon)) (*Daemon, error) {
```

Comment:
Make the old `NewDaemon/Client` functions call into the new ones instead of duplicating the code.